pkg/image/validation.go

```diff
@@ -64,6 +70,31 @@ func validateOperatingSystem(definition *Definition) error {
 	return nil
 }
 
+func validateKubernetes(definition *Definition) error {
+	if definition.Kubernetes.Version == "" {
+		// Not configured
+		return nil
+	}
+
+	supportedCNIs := []string{CNITypeNone, CNITypeCanal, CNITypeCalico, CNITypeCilium}
+
+	if definition.Kubernetes.CNI == "" {
+		log.Audit("Kubernetes CNI was not specified. Please set \"cni: none\" if you intend to use your own")
```

Comment:
This is an edge case which we can technically log to the file, however, the message will be written before writing the following error which might be counterintuitive. I settled for auditing it but I'm also fine with revisiting this if it doesn't make sense.

Reply:
Not 100% sure I follow, but I'm fine with this approach until we see how it plays out during usage.